src/pages/Dice.js

```diff
@@ -0,0 +1,335 @@
+import React, { useState, useEffect } from 'react'
+import { Link } from 'react-router-dom'
+import styles from './Dice.module.css'
+import { ReactComponent as Dice1 } from '../components/icons/dice1.svg'
+import { ReactComponent as Dice2 } from '../components/icons/dice2.svg'
+import { ReactComponent as Dice3 } from '../components/icons/dice3.svg'
+import { ReactComponent as Dice4 } from '../components/icons/dice4.svg'
+import { ReactComponent as Dice5 } from '../components/icons/dice5.svg'
+import { ReactComponent as Dice6 } from '../components/icons/dice6.svg'
+import { useToken } from '../components/hooks/useToken'
+import moment from 'moment'
+import DiceImg from '../img/dice.png'
+
+function Dice() {
+    const [bet, setBet] = useState(null)
+    const [isLoading, setLoading] = useState(false)
+    const [showHistory, setShowHistory] = useState(false)
+    const [historyData, setHistoryData] = useState([])
+    const [result, setResult] = useState(null)
+
+    const token = useToken()
+
+    const platform = window.navigator.userAgentData.mobile
+        ? 'Web-mobile'
+        : 'Web-desktop'
+
+    const getHistory = () => {
```

Comment:
Suggest moving to a separate service file with all fetches.

Reply:
After demo